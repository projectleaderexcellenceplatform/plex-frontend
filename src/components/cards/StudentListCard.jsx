import React, {useState} from "react";

import avtar from "../../assets/images/avtar.svg";
import info from "../../assets/images/icons/info.svg";

const StudentListCard = (props) => {
    const latestProject = [props.project];

    const [active, setActive] = useState(false);


    return (
        <div>
            {latestProject
                .map((item) => {
                    return (
                        <div
                            className={"project-Card_b d-flex justify-content-between align-items-center w-100 shadow rounded-10 px- py-3"}>
                            <div className="row align-items-center w-100">
                                <div className="col-5">
                                    <div className="row align-items-center">
                                        <div className="col-2">
                                            {/*<img/>*/}
                                        </div>
                                        <div className="col-10 ps-0">
                                            <p className="mb-0 f14 color1 fw800">
                                                {item.title}
                                            </p>
                                        </div>
                                    </div>
                                </div>
                            </div>

                            <div className={"d-flex justify-content-between"}>
                                {[1, 1, 1].map((priorityButton, i) => {
                                    return (
                                        <div
                                            key={i}
                                            onClick={() => {
                                               props.func(item.id, i)
                                            }}
                                            className={active ? "bg-color2 text-white card-list-student-checkbox" : "bg-color4 color5 card-list-student-checkbox"}
                                        >
                                            {i + 1}
                                        </div>
                                    )
                                })
                                }
                            </div>


                        </div>

                    )
                })}

        </div>

    );
};

export default StudentListCard;

//                                        onClick={() => setActive(prevActive => !prevActive)}

// {[1, 1, 1].map((item, i) => {
//     return (
//         <div
//             key={i}
//             onClick={() => {
//                 setActive(!active);
//             }}
//             className={active ? "bg-color2 text-white card-list-student-checkbox": "bg-color4 color5 card-list-student-checkbox"}
//         >
//             {i + 1}
//         </div>
//     )
// })
// }