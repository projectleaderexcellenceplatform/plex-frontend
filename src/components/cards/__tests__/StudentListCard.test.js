import React, { useState } from "react";

import {render, screen, cleanup} from "@testing-library/react";
import renderer from 'react-test-renderer'
import StudentListCard from "../StudentListCard";

afterEach(() => {
    cleanup();
})

test('matches snapshot', () => {
    const project = [
        {
            "categories": [
                {
                    "id": 1,
                    "name": "UX/UI"
                }
            ],
            "projects": [
                {
                    "id": 7,
                    "title": "Project7",
                    "shortDescription": "Omschrijving7",
                    "uploader": "uploader7",
                    "categories": [
                        {
                            "id": 1,
                            "name": "UX/UI"
                        }
                    ]
                },
                {
                    "id": 1,
                    "title": "Project1",
                    "shortDescription": "Omschrijving1",
                    "uploader": "uploader1",
                    "categories": [
                        {
                            "id": 1,
                            "name": "UX/UI"
                        }
                    ]
                }
            ]
        }
    ]

    const tree = renderer.create(<StudentListCard project={project}/>).toJSON();
    expect(tree).toMatchSnapshot();
});