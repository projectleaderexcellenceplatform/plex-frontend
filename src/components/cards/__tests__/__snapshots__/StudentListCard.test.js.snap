// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="project-Card_b d-flex justify-content-between align-items-center w-100 shadow rounded-10 px- py-3"
  >
    <div
      className="row align-items-center w-100"
    >
      <div
        className="col-5"
      >
        <div
          className="row align-items-center"
        >
          <div
            className="col-2"
          />
          <div
            className="col-10 ps-0"
          >
            <p
              className="mb-0 f14 color1 fw800"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="d-flex justify-content-between"
    >
      <div
        className="bg-color4 color5 card-list-student-checkbox"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="bg-color4 color5 card-list-student-checkbox"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="bg-color4 color5 card-list-student-checkbox"
        onClick={[Function]}
      >
        3
      </div>
    </div>
  </div>
</div>
`;
